teps/0047-pipeline-task-display-name.md

```diff
@@ -0,0 +1,147 @@
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A task in a Pipeline is currently represented in the UI using a field (`name`) that is meant to be
+machine readable, not human readable. There should be the addition of a new
+optional field (`displayName`) to describe Pipeline tasks that are human readable.
+
+## Motivation
+
+Simple version, humans want to read task names formatted for human's not computers, and they want
+to see the full task name rendered in their UI.
+
+
+The longer version, the end user of a Tekton pipeline can vary including but not limited to
+application developer, to security professional, to compliance officer,
+service reliability engineer, product manager, and beyond. The farther
+you move away from the application developer the more important human
+understanding of the CI/CD process is important.
+
+Currently if a product manager or auditor was to look at a Tekton Pipeline run rendered in UI
+that would be presented with a bunch of truncated machine readable task names, no spaces, no
+capitalization, and the entire task name isn't even readable beyond 20 or so characters.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a `Pipeline.tasks.*.displayName` (or similar) field that allows any text and
+  if provided will be rendered when displaying Tekton Pipeline in UI rather then the
+  machine readable `Pipeline.tasks.*.name` field.
+* Ideally when rendering this new display name it will be fully rendered rather then the
+  current truncation to 20 characters that is currently happening.
+
+### Non-Goals
+
+To be determined.
+
+### Use Cases (optional)
+
+Pipeline writers can specify a display name for any/all tasks, if supplied that display name
+will be rendered in UI rather then the machine readable `name` field value.
+
+#### Example
+```
+kind: PipelineRun
+spec:
+  pipelineSpec:
+    tasks:
+    - name: echo-message
+      displayName: Echo Message
+      taskSpec:
+        steps:
+        - name: echo
+          image: ubuntu
+          script: |
+            echo "Hi!"
+```
+
+## Requirements
+
+New display name filed would be optional and accept any unicode character
+
+## Proposal
+
+Add `Pipeline.tasks.*.displayName` and update Tekton rendering to use that field rather then
+`Pipeline.tasks.*.name` when provided.
+
+### Notes/Caveats (optional)
+
+None.
+
+### Risks and Mitigations
+
+If pursue always rendering the full display name, for extremly large stings could start to cause
+rendering issues. Mitigations would be to look out how other similar tools solve this problem.
```

Comment:
It would be great to nail down what the mitigation would be.  Is there an acceptable maximum length for the field?

Reply:
i dont have a good answer for this one. All I can do is compare against other tools in this space and know i get less information / less visually pleasing information, out of looking at the rendered pipeline in tekton then i do the exact same pipeline rendered in another tool such as jenkins. we have other tools yet to re-implement the same pipeline in to see what they render like but that is work still to be done. I will attach some screen shots to a comment.